File: src/javascript/send.js
var zmq = require('/usr/local/lib/node_modules/zmq'), 
	msgpack = require('/usr/local/lib/node_modules/msgpack');

if (process.argv.length < 5) {
	console.log("usage: ", process.argv[0], process.argv[1], " bind|connect port msg");
	process.exit(1);
}

var sender = zmq.socket('push');

if (process.argv[2] == "bind") {
	sender.bind(process.argv[3]);
}
else if (process.argv[2] == "connect") {
	sender.connect(process.argv[3]);
}
else {
	console.log("bind or connect");
	process.exit(1);
}

var packed = msgpack.pack(process.argv.slice(4))
sender.send(packed);

process.exit(0);

File: src/python/send.py

import zmq
import msgpack
import sys

if len(sys.argv) < 4:
	print "usage: ", sys.argv[0], " bind|connect port msg"
	exit(1)

context = zmq.Context()

sender = context.socket(zmq.PUSH)
if sys.argv[1] == "bind":
	sender.bind(sys.argv[2])
elif sys.argv[1] == "connect":
	sender.connect(sys.argv[2])
else:
	print "bind or connect"
	exit(1)

packed = msgpack.packb(sys.argv[3:], use_bin_type=True)
sender.send(packed)

sender.close()
context.term()

exit(0)
